src/pages/workspace/upgrade/WorkspaceUpgradePage.tsx

```diff
@@ -19,22 +21,51 @@ import UpgradeIntro from './UpgradeIntro';
 type WorkspaceUpgradePageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.UPGRADE>;
 
 function WorkspaceUpgradePage({route}: WorkspaceUpgradePageProps) {
+    const navigation = useNavigation();
     const styles = useThemeStyles();
     const policyID = route.params.policyID;
     const feature = Object.values(CONST.UPGRADE_FEATURE_INTRO_MAPPING).find((f) => f.alias === route.params.featureName);
     const {translate} = useLocalize();
     const [policy] = useOnyx(`policy_${policyID}`);
     const {isOffline} = useNetwork();
 
-    if (!feature || !policy) {
-        return <NotFoundPage />;
-    }
+    const isUpgraded = React.useMemo(() => isControlPolicy(policy), [policy]);
 
     const upgradeToCorporate = () => {
-        Policy.upgradeToCorporate(policy.id, feature.id);
+        if (!policy || !feature) {
```

Comment:
Do we not have a way in the UI to upgrade without having to click on a feature first?